crypto/init.c

```diff
@@ -630,7 +644,7 @@ int OPENSSL_init_crypto(uint64_t opts, const OPENSSL_INIT_SETTINGS *settings)
     if (opts & OPENSSL_INIT_LOAD_CONFIG) {
         int ret;
         CRYPTO_THREAD_write_lock(init_lock);
-        appname = (settings == NULL) ? NULL : settings->appname;
+        conf_settings = settings;
         ret = RUN_ONCE(&config, ossl_init_config);
```

Comment:
Perhaps conf_settings should be reset to NULL here?

Reply:
Sure, though the only way it ever gets used is via the above code path which always resets the static, but perhaps safer to reset after use.  I'll add that when the dust settles...